# Reasoning behind the >2.5 bet market. #
Data analysis of the PL data set and scope for use of the ML model.

Import the data and required libariers

In [1]:
#import
import pandas as pd


#read our data
matches= pd.read_csv("data_21_23.csv")


Create some standard data analysis functions

In [2]:
##A function to calculate the av goals of a team
total_goals=0
def calculate_avg_goals_per_game_scored(df, team_name):
    
    home_matches = df[(df['HomeTeam'] == team_name)]
    away_matches=df[(df['AwayTeam'] == team_name)]
    total_goals = home_matches['FTHG'].sum() + away_matches['FTAG'].sum()
    total_matches = home_matches.shape[0] +away_matches.shape[0]
    avg_goals_per_game = total_goals / total_matches
    return avg_goals_per_game

In [3]:
##A function to calculate the toal goals conceeded
def calculate_avg_goals_per_game_conceeded(df, team_name):
    
    home_matches = df[(df['HomeTeam'] == team_name)]
    away_matches=df[(df['AwayTeam'] == team_name)]
    total_goals = home_matches['FTAG'].sum() + away_matches['FTHG'].sum()
    total_matches = home_matches.shape[0] +away_matches.shape[0]
    avg_goals_per_game = total_goals / total_matches
    return avg_goals_per_game


In [4]:
 ##Over 3 goals scored in a game
def three_goal_games(df):
    high_scoring = df[(df['FTHG'] + df['FTAG']) >= 3]
    sum_high_scoring= high_scoring.shape[0]
    return sum_high_scoring

In [5]:
## Find number of matches played
def num_matches(df):
    return df.shape[0]


In [13]:
##average odds for >2.5 market
def av_high_scoring_odds(df):
    return df['AvgC>2.5'].mean()

Comparing the odds for >2.5 goal games to the proportion of games that are high scoring (>2.5 goals), we see that the margin is tight:

In [8]:
print(three_goal_games(matches))

405


In [10]:
print(num_matches(matches))

760


In [14]:
print(av_high_scoring_odds(matches))

1.8543684210526319


If we hypothetically bet £1 on every game to have >2.5 goals, then we would make (405 x 0.85) in profit and lose (760 - 405) in lose bets. This leaves us at -£8.98 or -1.18% of total wager. So a model need only be 1.19% better than random chance in order to make profit*


*Expected

Based purley on number of games scoring higher, the expected "fair" odds would be 1.877, and with a true odds average of 1.85, the house is only making a small edge on this market.